src/components/application_manager/src/hmi_state.cc

```diff
@@ -242,11 +242,7 @@ PhoneCallHmiState::PhoneCallHmiState(std::shared_ptr<Application> app,
     : HmiState(app, app_mngr, STATE_ID_PHONE_CALL) {}
 
 mobile_apis::HMILevel::eType PhoneCallHmiState::hmi_level() const {
-  using namespace mobile_apis;
-  if (HMILevel::INVALID_ENUM == parent_hmi_level()) {
-    return parent_hmi_level();
-  }
-  return std::max(parent_hmi_level(), max_hmi_level());
+  return parent_hmi_level();
```

Comment:
Really, you shouldn't be changing this function, `max_hmi_level` is the one you should be editing. That said, since it seems that we are removing any HMI level changes for this event, we should just be removing these overrides.

Both the `PhoneCallHmiState::hmi_level()` and `PhoneCallHmiState::max_hmi_level()` functions can be removed from this file and `hmi_state.h`

```
  PhoneCallHmiState(std::shared_ptr<Application> app,
                    const ApplicationManager& app_mngr);

-  mobile_apis::HMILevel::eType hmi_level() const OVERRIDE;
  mobile_apis::AudioStreamingState::eType audio_streaming_state()
      const OVERRIDE {
    return mobile_apis::AudioStreamingState::NOT_AUDIBLE;
  }

-  mobile_apis::HMILevel::eType max_hmi_level() const OVERRIDE;
  mobile_apis::AudioStreamingState::eType max_audio_streaming_state()
      const OVERRIDE {
    return audio_streaming_state();
```

Reply:
Hi, @jacobkeeler -san.
I took over #3227 from @Yuki-Kanadome-Nexty because he was off this project.
Thank you for your comment.
Is what you mean to remove PhoneCallHmiState::hmi_level() and PhoneCallHmiState::max_hmi_level() from hmi_state.h and hmi_state.cc?
Or, is it to remove PhoneCallHmiState::hmi_level() and PhoneCallHmiState::max_hmi_level() from hmi_state.h and to cancel  ​the modification of hmi_state.cc?

Reply:
@Akihiro-Miyazaki You would be removing these functions from both `hmi_state.h` and `hmi_state.cc`

Reply:
@jacobkeeler -san,
Thank you for your reply.
We confirmed your implementation and understood that there is no different behavior between current implementation and your one. If you have the same recognition, since the test of current implementation is completed, we would like to select current one. Is it possible?

Reply:
There is a difference in the implementation, currently max_hmi_level is restricted (meaning if you try to activate the app during the phone call event, it will still be restricted to BACKGROUND or LIMITED). Considering that this state is not supposed to affect the HMI level at all, I believe the only correct solution is to remove these overrides.

Reply:
In addition, the other reason these changes are incorrect is that `hmi_level` should ALWAYS take `max_hmi_level` into account. Your changes just remove the `max_hmi_level` factor from this method, which makes the state behave incorrectly.

To explain the difference:
- `hmi_level()` returns the HMI level of the app immediately after this event becomes active
- `max_hmi_level()` returns the highest HMI level that the app can enter while the state is active

Reply:
@jacobkeeler -san,
Sorry for delay our response, as we confirmed your suggestion.
We recognized that your suggestion means to remove the override function and to use the parent class function.
Is it correct?
However, since the parent class is defined as the virtual function, we modified to call the parent function on the override function like below.
If our recognition is wrong or you have any questions, could you please comment on us.
Thank you.

hmi_state.cc
```
- mobile_apis::HMILevel::eType PhoneCallHmiState::hmi_level() const {
-   using namespace mobile_apis;
-   if (HMILevel::INVALID_ENUM == parent_hmi_level()) {
-     return parent_hmi_level();
-   }
-   return std::max(parent_hmi_level(), max_hmi_level());
- }

- mobile_apis::HMILevel::eType PhoneCallHmiState::max_hmi_level() const {
-   using namespace helpers;
-   using namespace mobile_apis;

-   if (WindowType::WIDGET == window_type()) {
-     return std::max(HMILevel::HMI_FULL, parent_max_hmi_level());
-   }

-   auto expected = HMILevel::HMI_FULL;
-   if (is_navi_app() || is_mobile_projection_app()) {
-     expected = HMILevel::HMI_LIMITED;
-   } else if (is_media_app()) {
-     expected = HMILevel::HMI_BACKGROUND;
-   }

-   return std::max(expected, parent_max_hmi_level());
- }
```

hmi_state.h
```
  class PhoneCallHmiState : public HmiState {
   public:
    PhoneCallHmiState(std::shared_ptr<Application> app,
                     const ApplicationManager& app_mngr);

-  mobile_apis::HMILevel::eType hmi_level() const OVERRIDE;
+   mobile_apis::HMILevel::eType hmi_level() 
+       const OVERRIDE {
+     return HmiState::hmi_level();
+   }

    mobile_apis::AudioStreamingState::eType audio_streaming_state()
        const OVERRIDE {
      return mobile_apis::AudioStreamingState::NOT_AUDIBLE;
    }

-   mobile_apis::HMILevel::eType max_hmi_level() const OVERRIDE;
+   mobile_apis::HMILevel::eType max_hmi_level()
+       const OVERRIDE {
+     return HmiState::max_hmi_level();
+   }

    mobile_apis::AudioStreamingState::eType max_audio_streaming_state()
        const OVERRIDE {
      return audio_streaming_state();
    }
  };
```

Reply:
That's mostly correct, though I don't see a reason to explicitly call the parent function, you will see that there are other states which do not override the `hmi_level` and `max_hmi_level` functions (ex. `TTSHmiState`). I would just remove the function definitions from `PhoneCallHmiState` entirely:


hmi_state.cc
```
- mobile_apis::HMILevel::eType PhoneCallHmiState::hmi_level() const {
-   using namespace mobile_apis;
-   if (HMILevel::INVALID_ENUM == parent_hmi_level()) {
-     return parent_hmi_level();
-   }
-   return std::max(parent_hmi_level(), max_hmi_level());
- }

- mobile_apis::HMILevel::eType PhoneCallHmiState::max_hmi_level() const {
-   using namespace helpers;
-   using namespace mobile_apis;

-   if (WindowType::WIDGET == window_type()) {
-     return std::max(HMILevel::HMI_FULL, parent_max_hmi_level());
-   }

-   auto expected = HMILevel::HMI_FULL;
-   if (is_navi_app() || is_mobile_projection_app()) {
-     expected = HMILevel::HMI_LIMITED;
-   } else if (is_media_app()) {
-     expected = HMILevel::HMI_BACKGROUND;
-   }

-   return std::max(expected, parent_max_hmi_level());
- }
```

hmi_state.h
```
  class PhoneCallHmiState : public HmiState {
   public:
    PhoneCallHmiState(std::shared_ptr<Application> app,
                     const ApplicationManager& app_mngr);

-   mobile_apis::HMILevel::eType hmi_level() const OVERRIDE;

    mobile_apis::AudioStreamingState::eType audio_streaming_state()
        const OVERRIDE {
      return mobile_apis::AudioStreamingState::NOT_AUDIBLE;
    }

-   mobile_apis::HMILevel::eType max_hmi_level() const OVERRIDE;

    mobile_apis::AudioStreamingState::eType max_audio_streaming_state()
        const OVERRIDE {
      return audio_streaming_state();
    }
  };
```

Reply:
@jacobkeeler -san,
Thank you for your advice.
We could understand.
By the way, since we have been several times to propose the files but we were a first time to contribute the files this time, there are some unclear points.
If possible, please let us know how to pull the files from other members' pull request and to push the files to other members' pull request?

Reply:
Hello Akihiro-Miyazaki,

> If possible, please let us know how to pull the files from other members' pull request and to push the files to other members' pull request?

What you are describing are basic functions of git: branching and merging. Here is [a tutorial by Atlassian](https://www.atlassian.com/git/tutorials/using-branches) which may help. If you could be more specific about your goal I may be able to help more.

If you do not have permission to the sdl_core fork created by `Yuki-Kanadome-Nexty` you will need to request permission from them, or create your own fork and a new pull request.